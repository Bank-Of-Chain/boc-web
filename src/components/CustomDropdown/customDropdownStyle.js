import {
  defaultFont,
  infoColor,
  infoBoxShadow,
  successColor,
  successBoxShadow,
  warningColor,
  warningBoxShadow,
  dangerColor,
  dangerBoxShadow,
  roseColor,
  roseBoxShadow,
} from "@/assets/jss/material-kit-react.js";

const customDropdownStyle = (theme) => ({
  popperClose: {
    pointerEvents: "none",
  },
  dropdown: {
    borderRadius: "12px",
    border: "0",
    top: "100%",
    zIndex: "1000",
    minWidth: "160px",
    padding: "5px 0",
    margin: "2px 0 0",
    fontSize: "14px",
    textAlign: "left",
    listStyle: "none",
    background: "#292b2e",
    backgroundClip: "padding-box",
    boxShadow: "0 0 0px 1px rgba(161, 161, 161, 0.2)",
  },
  menuList: {
    padding: "0",
  },
  popperResponsive: {
    zIndex: "1200",
    [theme.breakpoints.down("sm")]: {
      zIndex: "1640",
      position: "static",
      float: "none",
      width: "auto",
      marginTop: "0",
      backgroundColor: "transparent",
      border: "0",
      boxShadow: "none",
      color: "black",
    },
  },
  dropdownItem: {
    ...defaultFont,
    fontSize: "13px",
    padding: "10px 20px",
    margin: "0 5px",
    borderRadius: "2px",
    position: "relative",
    transition: "all 150ms linear",
    display: "block",
    clear: "both",
    fontWeight: "400",
    height: "fit-content",
    color: "#A0A0A0",
    whiteSpace: "nowrap",
    minHeight: "unset",
  },
  blackHover: {
    "&:hover": {
      boxShadow:
        "0 4px 20px 0px rgba(0, 0, 0, 0.14), 0 7px 10px -5px rgba(33, 33, 33, 0.4)",
      backgroundColor: "#212121",
      color: "#fff",
    },
  },
  primaryHover: {
    "&:hover": {
      background: "#323438",
      color: "#A68EFE",
    },
  },
  infoHover: {
    "&:hover": {
      backgroundColor: infoColor,
      color: "#FFFFFF",
      ...infoBoxShadow,
    },
  },
  successHover: {
    "&:hover": {
      backgroundColor: successColor,
      color: "#FFFFFF",
      ...successBoxShadow,
    },
  },
  warningHover: {
    "&:hover": {
      backgroundColor: warningColor,
      color: "#FFFFFF",
      ...warningBoxShadow,
    },
  },
  dangerHover: {
    "&:hover": {
      backgroundColor: dangerColor,
      color: "#FFFFFF",
      ...dangerBoxShadow,
    },
  },
  roseHover: {
    "&:hover": {
      backgroundColor: roseColor,
      color: "#FFFFFF",
      ...roseBoxShadow,
    },
  },
  dropdownItemRTL: {
    textAlign: "right",
  },
  dropdownDividerItem: {
    margin: "5px 0",
    backgroundColor: "rgba(0, 0, 0, 0.12)",
    height: "1px",
    overflow: "hidden",
  },
  buttonIcon: {
    width: "24px !important",
    height: "24px !important",
  },
  dropdownTrigger: {
    "&.MuiButtonBase-root.MuiButton-root": {
      paddingRight: "8px",
    },
  },
  caret: {
    transition: "all 150ms ease-in",
    display: "inline-block",
    marginLeft: "4px",
    verticalAlign: "middle",
  },
  caretActive: {
    transform: "rotate(180deg)",
  },
  caretRTL: {
    marginRight: "4px",
  },
  dropdownHeader: {
    display: "block",
    padding: "0.1875rem 1.25rem",
    fontSize: "0.75rem",
    lineHeight: "1.428571",
    color: "#777",
    whiteSpace: "nowrap",
    fontWeight: "inherit",
    marginTop: "10px",
    minHeight: "unset",
    "&:hover,&:focus": {
      backgroundColor: "transparent",
      cursor: "auto",
    },
  },
  noLiPadding: {
    padding: "0",
  },
});

export default customDropdownStyle;
