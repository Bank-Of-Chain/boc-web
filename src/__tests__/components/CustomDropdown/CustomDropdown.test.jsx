import React from 'react'
import { render } from '@testing-library/react'
import CustomDropdown from '@/components/CustomDropdown/CustomDropdown'
import { describe, expect, it } from 'vitest'

describe('xxxxxxxx', () => {
  it('CustomDropdown Component Render', () => {
    const { asFragment } = render(<CustomDropdown hoverColor="primary" buttonProps={{ className: 'abc' }} buttonText="abc" buttonIcon="abc" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
