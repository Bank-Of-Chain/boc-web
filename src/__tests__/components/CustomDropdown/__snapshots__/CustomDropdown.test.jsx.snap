// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomDropdown Component Render 1`] = `
<DocumentFragment>
  <div>
    <div>
      <button
        aria-haspopup="true"
        aria-label="Notifications"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-22 abc makeStyles-dropdownTrigger-16"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root makeStyles-buttonIcon-15"
          >
            abc
          </span>
          abc
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-caret-17"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
